composites/OnboardingWizard/helpers/postJSON.js

```diff
@@ -46,46 +41,60 @@ let postJSONFetch = ( url, data ) => {
 		}
 	);
 };
+
 /**
  * Handles JSON request the jQuery way.
  *
  * @param {string} url  The endpoint to send the data to.
+ * @param {object} headers Object containing the headers for the request.
  * @param {Object} data The JSON object to send to the server.
  *
  * @returns {Promise} A Promise, if the request is successful the promise is resolved, else it's rejected.
  */
-let postJSONjQuery = ( url, data ) => {
-	let promise = new Promise( ( resolve, reject )=> {
-		jQuery.post( { url, dataType: "json", data } )
-		      .done(
-			      ( response ) => {
-				      resolve( response );
-			      }
-		      )
-		      .fail(
-			      () => {
-				      reject( "Wrong request" );
-			      }
-		      );
+let postJSONjQuery = ( url, headers, data ) => {
```

Comment:
Since you use PUT requests instead of POST request, you should rename some variables in this file and also the filename.